src/NHibernate.Test/StaticProxyTest/StaticProxyFactoryFixture.cs

```diff
@@ -158,7 +158,7 @@ public void VerifyProxyForClassWithInternalInterface()
 		}
 
 		[Test]
-		public void VerifyProxyOnInterface()
+		public void VerifyProxyForClassWithAdditionalInterface()
```

Comment:
This test generates a proxy class solely based on the interfaces. It is not actually a proxy of the class. Going to that name is going back to the confusion. Even worse combine with the other renaming.

We now have:

- `VerifyProxyForClassWithInterface`: does test a `class proxy: PublicInterfaceTestClass, INHibernateProxy, IPublic`.
- `VerifyProxyForClassWithAdditionalInterface`: does actually test a `class proxy: INHibernateProxy, IPublic`, without any direct relationship to `PublicInterfaceTestClass`. `Additional` is far from telling this.
   The fact that obtaining this requires to add the interface in the interface list of the proxy factory is a (bad) implementation detail. It seems way more important to me to describe the kind of proxy we test, rather than the way we obtain such proxy.